src/components/game/game-object/GameObject.js

```diff
@@ -1,16 +1,17 @@
 import React, { useEffect, useRef, useContext } from 'react';
 import { GlobalContext } from '../../../context/GlobalState';
-import { useCollider } from '../../../hooks/collider';
+import { useCollider } from './useCollider';
 import DraggableSprite from './DraggableSprite';
 import Collider from './Collider';
 
-export const GameObject = ({ id, gameObjects, images }) => {
-  const { zPositions } = useContext(GlobalContext);
+export const GameObject = ({ id, images }) => {
+  const { zPositions, gameObjects } = useContext(GlobalContext);
   const self = useRef(gameObjects[id]).current;
   const { handleCollision } = useCollider();
 
   useEffect(() => {
-    gameObjects[id].z = zPositions[self.type];
+    // -1 because when getting from stack we want newly spawned to be under one that is taken
+    gameObjects[id].z = zPositions[self.type] - 1;
```

Comment:
To nie spowoduje że te nowe w końcu będą za nisko?

Reply:
tzn?
to jest ustawione pod stosy i dziala dobrze, problem bedzie jak bedziemy spawnowac nowe z palca, np jak bedziemy pojawiac sklepowe itemy, wtedy one beda ponizej obecnego najwyzszego. to sie ogarnie jak dojdziemy do spawnowania sklepowych, na razie dziala dobrze. trzeba bedzie dodac jakiegos ifa np zrodlo spawnu czy jest stos czy menu spawnu.